### DISCUSSION TITLE
Vector Agent as Library

### DISCUSSION THREAD

**Discussion by ronaldpetty** [vectordotdev/vector#19776] [General]
### A note for the community

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Problem

Does Vector have an option to be loaded as a library instead of being used as a agent? I ask due to some architectures moving away from the sidecar approach, things like proxyless grpc or eBPF variants of tools (like Istio). There of course would be pros and cons of this approach. 

I noticed other tickets (for Vector developers) discussing use of crates (or not) to help with build efforts. Maybe a subset could be used in such a form.

Any thoughts (or links to prior conversations - I didn't find any) appreciated.

### Version

vector 0.35.0 (x86_64-unknown-linux-gnu e57c0c0 2024-01-08 14:42:10.103908779)

**Comment by jszwedko** [comment]
Hi @ronaldpetty !

It is possible to use Vector as a library (we've heard of others doing it), but it's not an explicit design goal at the moment so it's not well documented how to go about this.
